components/fxa-client/src/internal/oauth.rs

```diff
@@ -327,17 +328,27 @@ impl FirefoxAccount {
         resp: OAuthTokenResponse,
         scoped_keys_flow: Option<ScopedKeysFlow>,
     ) -> Result<()> {
+        let sync_scope_requested = resp.scope.split(' ').any(|s| s == scopes::OLD_SYNC);
```

Comment:
Would this be `sync_scope_granted`, I [**think** oauth allows the server to not grant a scope that was requested](https://datatracker.ietf.org/doc/html/rfc6749#section-3.3) (not sure how true this is for fxa and the sync scope thou), so I think what we would be logging is a server bug, where it tells us that it is granting us a sync scoped_key, then it is missing from the payload

Not sure how to validate the case where we request a scope, and the oauth server simply rejects and decides not to grant us the scoped key (by omitting it from the `scopes` in the response 🤷 Either way I think logging is the way to go

Reply:
Good point.  It would be really nice to track which scopes we requested, but I couldn't find a simple way to do that.  If this doesn't flag issues, then that could be the next step.

I'll update the variable name too.